From 84c6964331569fd605ce93ff94a910904b145341 Mon Sep 17 00:00:00 2001
From: Patrick Ohly <patrick.ohly@intel.com>
Date: Thu, 13 Aug 2015 17:57:32 +0200
Subject: [PATCH 4/4] calc_evm_hmac/hash: avoid fd leak when ioctl fails

When opening the file succeeds but ioctl() then fails, the file must
be closed before returning.

Signed-off-by: Patrick Ohly <patrick.ohly@intel.com>
---
 src/evmctl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/evmctl.c b/src/evmctl.c
index 55009c0..2b3589c 100644
--- a/src/evmctl.c
+++ b/src/evmctl.c
@@ -341,6 +341,7 @@ static int calc_evm_hash(const char *file, unsigned char *hash)
 		}
 		if (ioctl(fd, FS_IOC_GETVERSION, &generation)) {
 			log_err("ioctl() failed\n");
+			close(fd);
 			return -1;
 		}
 		close(fd);
@@ -906,6 +907,7 @@ static int calc_evm_hmac(const char *file, const char *keyfile, unsigned char *h
 		}
 		if (ioctl(fd, FS_IOC_GETVERSION, &generation)) {
 			log_err("ioctl() failed\n");
+			close(fd);
 			goto out;
 		}
 		close(fd);
-- 
2.1.4

